lib/runtime/scheduler.js

```diff
@@ -12,33 +16,72 @@ var Scheduler = Base.extend({
     },
 
     schedule: function(time, callback) {
+        if (this.stopped === true) {
+            throw errors.runtimeError('RT-CANNOT-SCHEDULE-AFTER-STOP');
+        }
+
         this.queue.push({ms:time, callback:callback});
-        if (this.timer) { clearTimeout(this.timer); }
-        this._run_after(0);
+
+        if (this.running === true) {
+            if (this.timer !== null) {
+                clearTimeout(this.timer);
+                this.timer = null;
+            }
+            this._run_after(0);
+        }
+    },
+
+    schedule_every: function(every, callback) {
+        var self = this;
+
+        if (this.stopped === true) {
+            throw errors.runtimeError('RT-CANNOT-SCHEDULE-AFTER-STOP');
+        }
+
+        var job = function() {
+            if (self.running === true) {
+                callback();
+                self.schedule(Date.now() + every, job);
```

Comment:
Seems like there would be an error if the execution of the callback caused stop to be called. Perhaps we want to have a second check for `self.running === true` before calling schedule again?